src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -116,14 +111,55 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 		}
 	}
 
-	public static transformData(rows: string[][], columns: any[]): { [key: string]: string }[] {
+	private static createIconColumn(col: azdata.TableColumn): Slick.Column<any> {
+		return <Slick.Column<any>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: iconCssFormatter,
+			filterable: false
+		};
+	}
+
+	private static createTextColumn(col: azdata.TableColumn): Slick.Column<any> {
+		return <Slick.Column<any>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: textFormatter,
+		};
+	}
+
+
+
+	public static transformData(rows: (string | azdata.IconColumnCellValue)[][], columns: any[], iconCssMap?: { [iconKey: string]: string }): { [key: string]: string | { text: string, ariaLabel: string } }[] {
 		if (rows && columns) {
+			let iconCssCache = iconCssMap ?? [];
 			return rows.map(row => {
-				let object: { [key: string]: string } = {};
+				let object: { [key: string]: string | { text: string, ariaLabel: string } } = {};
 				if (row.forEach) {
 					row.forEach((val, index) => {
 						let columnName: string = (columns[index].value) ? columns[index].value : <string>columns[index];
-						object[columnName] = val;
+						if (val['icon']) {
```

Comment:
since the val can be string or azdata.IconColumnCellValue, if you do 
```
if ('icon' in val) {

}
```
the compiler will know the type is IconColumnCellValue and you don't have to to the type cast.

Reply:
if I write `if ('icon' in val)` the compiler returns error 
`The right-hand side of an 'in' expression must be of type 'any', an object type or a type parameter.`

If I try convert to (<azdata.IconColumnCellValue>val) - the runtime will throw error trying to use 'in' with string value.

Reply:
Or even make a helper function like I mentioned above similar to https://github.com/microsoft/azuredatastudio/blob/main/src/sql/base/browser/ui/table/formatters.ts#L49